Formula/nerdctl.rb

```diff
@@ -0,0 +1,21 @@
+class Nerdctl < Formula
+  desc "ContaiNERD CTL - Docker-compatible CLI for containerd"
+  homepage "https://github.com/containerd/nerdctl"
+  url "https://github.com/containerd/nerdctl/archive/refs/tags/v0.15.0.tar.gz"
+  sha256 "7caf808a271d47ff069875cd49ceed02cf3fd3c0c334df2d89b3a031b0e6d041"
+  license "Apache-2.0"
+  head "https://github.com/containerd/nerdctl.git", branch: "master"
+
+  depends_on :linux
+  depends_on "go" => :build
+
+  def install
+    ldflags = "-s -w -X github.com/containerd/nerdctl/pkg/version.Version=#{version}"
+    system "go", "build", *std_go_args(output: bin/"nerdctl", ldflags: ldflags),
+      "./cmd/nerdctl"
+  end
+
+  test do
+    system "nerdctl", "-v"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Thanks for the help! I added a test with a failing output (as it requires containerd to run)